test/clients/retry_client.go

```diff
@@ -38,55 +42,56 @@ func retry(ctx context.Context, f func() error) error {
 			return err
 		}
 		err := f()
-		if delay, needDelay := apiutil.DelayIfRateLimited(err); needDelay {
+
+		if meta.IsNoMatchError(err) {
 			select {
 			case <-ctx.Done():
 				return ctx.Err()
-			case <-time.After(delay):
+			case <-time.After(5): //TODO: retrieve delay
```

Comment:
@shomron looks like `DelayIfRateLimited` is removed now, any ideas what we can use here for delay?

Reply:
hrm. A few options I see are:

1. Implement our own rate limiter here to avoid the busy loop.
2. Thread the same limiter here (the `RetryClient` initialization) and to the `DynamicRestMapper` (with `WithLimiter`). Not sure if this would work accurately since we don't have access to the mapper's reservation.
3. Implement a different `DynamicRestMapper` with the old behavior.

I vote the first option. Thoughts?

Reply:
Thanks @shomron, updated to use `limiter.Wait(ctx)`. Let me know what you think